crates/ruff_python_formatter/src/expression/string.rs

```diff
@@ -728,14 +728,15 @@ fn normalize_string(input: &str, quotes: StringQuotes, is_raw: bool) -> Cow<str>
 /// to the next multiple of 8. This is effectively a port of
 /// [`str.expandtabs`](https://docs.python.org/3/library/stdtypes.html#str.expandtabs),
 /// which black [calls with the default tab width of 8](https://github.com/psf/black/blob/c36e468794f9256d5e922c399240d49782ba04f1/src/black/strings.py#L61)
-fn count_indentation_like_black(line: &str, tab_width: TabWidth) -> TextSize {
+/// with the exception that Ruff uses [`IndentWidth`] for the tab width.
+fn count_indentation_like_black(line: &str, indent_width: IndentWidth) -> TextSize {
     let mut indentation = TextSize::default();
+    let indent_width = indent_width.value();
     for char in line.chars() {
         if char == '\t' {
             // Pad to the next multiple of tab_width
-            indentation += TextSize::from(
-                tab_width.value() - (indentation.to_u32().rem_euclid(tab_width.value())),
-            );
+            indentation +=
```

Comment:
Both black and [cpython itself](https://github.com/python/cpython/blob/b303d3ad3e80e1d9b3befe6650f61f38b72179a4/Lib/inspect.py#L884) have this hardcoded to 8, e.g.
```python
print(inspect.cleandoc('''
def a():
    """doc line 1
    doc line 2
\tdoc line 3
    """
'''))
```
expands to
```python
def a():
    """doc line 1
    doc line 2
        doc line 3
    """
```
Given that this is a cpython choice, i don't think we should have an option to overwrite it.

Reply:
Thanks for bringing this up. I wasn't aware of this (although this PR doesn't change the configurable nature of tab width).

If that's the case, do we foresee any issues when users use `indent-style: Tab` with a `tab-width` other than 8? Is this reason enough to disallow the  `indent-style: Tab` option altogether (and enforce spaces with various widths)?

Reply:
I've tested the semantics:

```python
source = '''
class A:
\tdef a():
\t\t"""doc line 1
\t\tdoc line 2
\t\t  doc line 3
\t\t"""

def a():
\t"""doc line 1
\tdoc line 2
\t  doc line 3
\t"""

class A2:
\tdef a():
\t\t"""doc line 1
\t\t\tdoc line 2
\t\t  doc line 3
\t\t"""

def a2():
\t"""doc line 1
\t\tdoc line 2
\t  doc line 3
\t"""
'''
exec(source)
docs = [A.a.__doc__, a.__doc__, A2.a.__doc__, a2.__doc__]
for doc in docs:
    print(f"---\n{inspect.cleandoc(doc)}")
```
prints
```
---
doc line 1
doc line 2
  doc line 3
---
doc line 1
doc line 2
  doc line 3
---
doc line 1
      doc line 2
doc line 3
---
doc line 1
      doc line 2
doc line 3
```

I've also checked that 
```python
source2 = '''
def f():
 \tprint()
 \tprint()

def f():
              \t          \t     print()
              \t          \t     print()
'''
exec(source2)
```
passes. Python seems to not care about how you mix tabs and spaces as long as you're consistent.

Inside of docstrings, `cleandoc` doesn't mind inconsistent indentation and will just pad to multiples of 8:
```python
source2 = '''
class A:
\tdef a():
\t\t"""doc line 1
 \t    \t doc line 2
\t \t  doc line 3
\t\t"""
'''
exec(source2)
print(f"---\n{inspect.cleandoc(A.a.__doc__)}")
```

My suggestion would be to special case `count_indentation_like_black` by hard code padding to 8 and converting to spaces in docstrings.

Reply:
Thanks for taking a closer look at this. I updated the PR to hardcode 8 for the normalization.